index.js

```diff
@@ -0,0 +1,18 @@
+function openNav() {
+  document.getElementById("mySidepanel").style.width = "200px";
+}
+
+function closeNav() {
+  document.getElementById("mySidepanel").style.width = "0";
+}
+
+const tasks = [];
+const addTaskButton = document.getElementById("add-task-button");
+addTaskButton.addEventListener("click", () => {
+  const taskTitle = document.getElementById("task-title").value;
+  tasks.push({
+    id: tasks.length + 1,
+    title: taskTitle,
+  });
+  console.log(tasks);
+});
```

Comment:
```suggestion
const tasks = [];
const addTaskButton = document.getElementById("add-task-button");
const taskTitleInput = document.getElementById("task-title-input");

addTaskButton.addEventListener("click", () => {
  tasks.push({
    id: tasks.length + 1,
    title: taskTitleInput.value,
  });
  
  console.log(tasks);
});
```

Reply:
So basically, the javascript's selectors can be out of the event's scope. The reason is that whenever the code uses `document.querySelector("...")` it searches the element in the DOM and find it again. So, putting that selector inside the event is not very efficient. Instead, to get the the current value form the input we can just use `.value`.

Reply:
So brilliant. I'll fix it